replication/logic/replication_logic.go

```diff
@@ -469,29 +483,23 @@ func (fs *Filesystem) doPlanning(ctx context.Context) ([]*Step, error) {
 			return nil, conflict
 		}
 
+		if initialSend {
+			// Add a nil path to the beginning so our loop over each step starts with a nil...FIRSTSNAPSHOT step.
+			// The nil `from` is what indicates the send is not incremental.
+			path = append([]*pdu.FilesystemVersion{nil}, path...)
```

Comment:
I'm not a big fan of this as it creates a hidden assumption in your slice. Where is the code that expects this nil?

Reply:
The code used to look like this in the "initial send" case (used to be called `len(..) == 1`):

```

			steps = append(steps, &Step{
				parent:   fs,
				sender:   fs.sender,

				from:    nil,
				to:      path[0],
				encrypt: fs.policy.EncryptedSend,
			})
```

seen in the diff below.

Reply:
I see. It feels hacky tbh. Personally, I'd go for this:

```
steps = make([]*Step, 0, len(path)) // shadow
if initialSend {
	steps = append(steps, &Step{
			parent:   fs,
			sender:   fs.sender,
			receiver: fs.receiver,

			from:    nil,
			to:      path[0],
			encrypt: fs.policy.EncryptedSend,
	})
}

for i := 0; i < len(path)-1; i++ {
	steps = append(steps, &Step{
		parent:   fs,
		sender:   fs.sender,
		receiver: fs.receiver,

		from:    path[i],
		to:      path[i+1],
		encrypt: fs.policy.EncryptedSend,
	})
}
```

or with something like this:

```
steps = make([]*Step, 0, len(path)) // shadow

appendStep := func(a, b *Step) {
	steps = append(steps, &Step{
		parent:   fs,
		sender:   fs.sender,
		receiver: fs.receiver,

		from:    a,
		to:      b,
		encrypt: fs.policy.EncryptedSend,
	})
}

if initialSend {
	appendStep(nil, path[0])
}

for i := 0; i < len(path)-1; i++ {
	appendStep(path[i], path[i+1])
}
```

Reply:
I like @kalbasit 's second suggestion